cli/bench_command.go

```diff
@@ -66,305 +62,540 @@ type benchCmd struct {
-	bench.Flag("pull", "Use a shared durable explicitly acknowledged JS pull consumer rather than individual ephemeral consumers").UnNegatableBoolVar(&c.pull)
-	bench.Flag("push", "Use a shared durable explicitly acknowledged JS push consumer with a queue group rather than individual ephemeral consumers").UnNegatableBoolVar(&c.pushDurable)
-	bench.Flag("consumerbatch", "Sets the batch size for the JS durable pull consumer, or the max ack pending value for the JS durable push consumer").Default("100").IntVar(&c.consumerBatch)
-	bench.Flag("pullbatch", "Sets the batch size for the JS durable pull consumer, or the max ack pending value for the JS durable push consumer").Hidden().Default("100").IntVar(&c.consumerBatch)
-	bench.Flag("subsleep", "Sleep for the specified interval before sending the subscriber acknowledgement back in --js mode, or sending the reply back in --reply mode,  or doing the next get in --kv mode").Default("0s").DurationVar(&c.subSleep)
-	bench.Flag("pubsleep", "Sleep for the specified interval after publishing each message").Default("0s").DurationVar(&c.pubSleep)
-	bench.Flag("history", "History depth for the bucket in KV mode").Default("1").Uint8Var(&c.history)
-	bench.Flag("multisubject", "Multi-subject mode, each message is published on a subject that includes the publisher's message sequence number as a token").UnNegatableBoolVar(&c.multiSubject)
-	bench.Flag("multisubjectmax", "The maximum number of subjects to use in multi-subject mode (0 means no max)").Default("100000").IntVar(&c.multiSubjectMax)
-	bench.Flag("retries", "The maximum number of retries in JS operations").Default("3").IntVar(&c.retries)
-	bench.Flag("dedup", "Sets a message id in the header to use JS Publish de-duplication").Default("false").UnNegatableBoolVar(&c.deDuplication)
-	bench.Flag("dedupwindow", "Sets the duration of the stream's deduplication functionality").Default("2m").DurationVar(&c.deDuplicationWindow)
+
+	addCommonFlags := func(f *fisk.CmdClause) {
+		f.Flag("clients", "Number of concurrent clients").Default("1").IntVar(&c.numClients)
+		f.Flag("msgs", "Number of messages to publish or subscribe to").Default("100000").IntVar(&c.numMsg)
+		f.Flag("no-progress", "Disable progress bar").UnNegatableBoolVar(&c.noProgress)
+		f.Flag("csv", "Save benchmark data to CSV file").StringVar(&c.csvFile)
+		f.Flag("size", "Size of the test messages").Default("128").StringVar(&c.msgSizeString)
+		// TODO: support randomized payload data
+	}
+
+	addPubFlags := func(f *fisk.CmdClause) {
+		f.Flag("multisubject", "Multi-subject mode, each message is published on a subject that includes the publisher's message sequence number as a token").UnNegatableBoolVar(&c.multiSubject)
+		f.Flag("multisubjectmax", "The maximum number of subjects to use in multi-subject mode (0 means no max)").Default("100000").IntVar(&c.multiSubjectMax)
+		f.Flag("sleep", "Sleep for the specified interval after publishing each message").Default("0s").DurationVar(&c.sleep)
```

Comment:
helps for these to do something like `PlaceHolder("DURATION")`, in fact using place holder can really improve the overall output of help - I dont use it nearly enough

Reply:
Done